src/main/java/org/openstreetmap/atlas/checks/flag/CheckFlag.java

```diff
@@ -302,17 +326,79 @@ public String getInstructions()
     }
 
     /**
-     * @return a list of {@link GeoJsonBuilder.LocationIterableProperties} representing all flagged
+     * Concatenates lists of {@link GeoJsonBuilder.GeometryWithProperties}
+     *
+     * @return a list of {@link GeoJsonBuilder.GeometryWithProperties} representing all flagged
      *         geometries
      */
-    public List<GeoJsonBuilder.LocationIterableProperties> getLocationIterableProperties()
+    public List<GeoJsonBuilder.GeometryWithProperties> getLocationIterableProperties()
     {
-        return this.flaggedObjects.stream()
-                .map(flaggedObject -> new GeoJsonBuilder.LocationIterableProperties(
-                        flaggedObject.getGeometry(), flaggedObject.getProperties()))
+
+        return Stream
+                .concat(getListOfGeoJsonObjectsForFlaggedRelation().stream(),
+                        this.flaggedObjects.stream()
+                                .map(flaggedObject -> new GeoJsonBuilder.GeometryWithProperties(
+                                        flaggedObject.getGeometry(),
+                                        new HashMap<>(flaggedObject.getProperties()))))
                 .collect(Collectors.toList());
     }
 
+    /**
+     * Creates geojson objects for the members of the flagged relations.
+     *
+     * @return a list of {@link GeoJsonBuilder.GeometryWithProperties} representing all flagged
+     *         geometries of FlaggedRelation
+     */
+    public List<GeoJsonBuilder.GeometryWithProperties> getListOfGeoJsonObjectsForFlaggedRelation()
+    {
+        final List<GeoJsonBuilder.GeometryWithProperties> locationIterablePropertyList = new ArrayList<>();
+        final Iterator<FlaggedRelation> iterator = this.flaggedRelations.iterator();
+        while (iterator.hasNext())
+        {
+
+            // Get flattened members of relation as a RelationMemberList
+            final RelationMemberList relationMembers = iterator.next()
+                    .getFlattenedRelationMembers();
+            for (final RelationMember relationMember : relationMembers)
+            {
+                final AtlasEntity entity = relationMember.getEntity();
+                final Map<String, Object> flaggedObjectProperties = new HashMap<>();
+                final JsonArray roles = new JsonArray();
+                FlaggedObject flaggedObject = null;
+
+                if (entity instanceof LocationItem)
+                {
+                    final FlaggedPoint flaggedPoint = new FlaggedPoint((LocationItem) entity);
+                    flaggedObjectProperties.putAll(flaggedPoint.getProperties());
+                    flaggedObject = flaggedPoint;
+                }
+                // If edge, consider only the master edges
+                else if (Edge.isMasterEdgeIdentifier(entity.getIdentifier()))
+                {
+                    // Consider only master edges
+                    if (Edge.isMasterEdgeIdentifier(entity.getIdentifier()))
```

Comment:
Looks like you're checking if Edge.isMaterEdge twice

Reply:
Oops!